crates/store/src/db/models/queries/nullifiers.rs

```diff
@@ -23,23 +23,49 @@ use crate::db::models::conv::{SqlTypeConvert, nullifier_prefix_to_raw_sql};
 use crate::db::models::utils::{get_nullifier_prefix, vec_raw_try_into};
 use crate::db::{NullifierInfo, schema};
 
-/// Returns nullifiers filtered by prefix and block creation height.
+/// Returns nullifiers filtered by prefix within a block number range.
 ///
-/// Each value of the `nullifier_prefixes` is only the `prefix_len` most significant bits
-/// of the nullifier of interest to the client. This hides the details of the specific
-/// nullifier being requested. Currently the only supported prefix length is 16 bits.
+/// Each value of the `nullifier_prefixes` is only the `prefix_len` most significant bits of the
+/// nullifier of interest to the client. This hides the details of the specific nullifier being
+/// requested. Currently the only supported prefix length is 16 bits.
 ///
 /// # Returns
 ///
-/// A vector of [`NullifierInfo`] with the nullifiers and the block height at which they were
+/// Range and pagination semantics:
+/// - Both `block_from` and `block_to` are inclusive bounds.
+/// - To keep responses ≤ ~2MB, an internal row limit is enforced. When the limit is hit and the
+///   result spans multiple blocks, the last block in the page is dropped entirely, and
+///   `last_block_included` is set to the block number immediately before that dropped block.
+///   Clients should resume with `block_from = last_block_included + 1`.
+/// - If all rows belong to a single block and hit the limit, this function returns an empty
+///   `nullifiers` page with `last_block_included = that_block - 1`. Intra-block pagination is not
+///   supported; callers may need to narrow the range.
```

Comment:
A block can have a maximum of $2^{16}$ nullifiers. So, if each nullifier is 36 bytes, at most a single block can take is 2.25MB (and this will return all nullifiers in a block). While hitting this limit is highly unlikely, I wonder if we should bump up the limit to something like 2.5MB to make sure this situation never happens.